spyder_kernels/console/start.py

```diff
@@ -87,33 +86,21 @@ def is_anaconda():
     return is_conda
 
 
-def exec_in_env(conda_root, envname, *command):
-    """
-    Run the standard conda activation script.
-    Taken from https://github.com/Anaconda-Platform/nb_conda_kernels/blob/
-    master/nb_conda_kernels/runner.py#L13
-    """
-    if sys.platform.startswith('win'):
-        print(sys.stdin.encoding, sys.stdout.encoding, sys.stderr.encoding,
-              locale.getpreferredencoding())
-        activate = os.path.join(conda_root, 'Scripts', 'activate.bat')
-        activator = subprocess.list2cmdline(['call', activate, envname])
-        ecomm = [os.environ['COMSPEC'], '/S', '/U', '/C', '@echo', 'off', '&&',
-                 'chcp', '&&', 'call', 'activate', envname,
-                 '&&'] + list(command)
-        subprocess.Popen(ecomm).wait()
-    else:
-        command = ' '.join(quote(c) for c in command)
-        activate = os.path.join(conda_root, 'bin', 'activate')
-        ecomm = ". '{}' '{}' && exec {}".format(activate, envname, command)
-        ecomm = ['sh' if 'bsd' in sys.platform else 'bash', '-c', ecomm]
-        os.execvp(ecomm[0], ecomm)
+def get_conda_root_path():
+    """Get conda root path using 'conda info --json'."""
+    conda_info = check_output('conda info --json')
+    conda_info_json = json.loads(conda_info)
+    return conda_info_json['env_vars']['CONDA_ROOT']
```

Comment:
Just is case is better to also use `dict.get` and add ifs to act in a more safe maner